c/lib.c

```diff
@@ -38,6 +38,109 @@ extern void *malloc(size_t length);
 #endif
 #endif
 
+// On Apple Silicon, `mrs` is not allowed in user-space, so we need to use the `sysctl` API.
+#if defined(__APPLE__) && defined(__MACH__)
+#define SZ_APPLE 1
+#include <sys/sysctl.h>
+#endif
+
+#if defined(__linux__)
+#define SZ_LINUX 1
+#endif
+
+SZ_INTERNAL sz_capability_t sz_capabilities_arm(void) {
+#if defined(SZ_APPLE)
+
+    // On Apple Silicon, `mrs` is not allowed in user-space, so we need to use the `sysctl` API.
+    uint32_t supports_neon = 0, supports_fp16 = 0, supports_bf16 = 0, supports_i8mm = 0;
+    size_t size = sizeof(supports_neon);
+    if (sysctlbyname("hw.optional.neon", &supports_neon, &size, NULL, 0) != 0) supports_neon = 0;
+    if (sysctlbyname("hw.optional.arm.FEAT_FP16", &supports_fp16, &size, NULL, 0) != 0) supports_fp16 = 0;
+    if (sysctlbyname("hw.optional.arm.FEAT_BF16", &supports_bf16, &size, NULL, 0) != 0) supports_bf16 = 0;
+    if (sysctlbyname("hw.optional.arm.FEAT_I8MM", &supports_i8mm, &size, NULL, 0) != 0) supports_i8mm = 0;
```

Comment:
The new M4 macs should support SVE and SVE2, but I don't know hot to check this. Do you have recent Apple hardware?

Reply:
sadly, I don't have access to the M4s yet :( I can check with friends or colleagues to see if 
they do and  can get the right sysctl property names for SVE/SVE2 checks

Reply:
I have an M4 iPad, we can probably try porting the SwiftSemanticSearch demo to iPad, integrating SimSIMD and testing on device. Can try next week.